# Go语言错误



## 错误的解决

​	在go语言中，错误不能使用try...catch...finally来解决。而是需要使用到defer 、recover及painc。解决

```go
func test(){
  //使用 defer+recover处理err
	defer func() {
		err := recover()
		if err != nil {
			fmt.Println(err)
		}
	}()
	a := 10
	b := 0
	i := a/b	//err发生位置
	fmt.Println(i)
}

func main() {
	test()
	fmt.Println("最后执行")
}
```

## 自定义错误

​	使用errors.New("错误信息")去定义错误